telink: tlx: add tl7218 board (Tercel A1) on BLE release v4.0.4.2

damien0x0023 opened the pull request:
- add new board tl7218x.
- update the TLX platform for tl7218x.
- update Zephyr TLX drivers for tl7218x.
- update hal_telink revision into west.yml.
- update the workflow for tl7218x.

andriy-bilynskyy commented:
The following west manifest projects have been modified in this Pull Request:

| Name | Old Revision | New Revision | Diff |
| ---- | ------------ | ------------ |------|
| hal_telink | https://github.com/telink-semi/hal_telink/commit/54c4302c9038d2cf412348400bcd039f5674da91 | https://github.com/telink-semi/hal_telink/commit/359564ebb09bf384b1d3d017cc06250052d813c6 (develop) | [telink-semi/hal_telink@54c4302c..359564eb](https://github.com/telink-semi/hal_telink/compare/54c4302c9038d2cf412348400bcd039f5674da91..359564ebb09bf384b1d3d017cc06250052d813c6) |

*Note: This message is automatically posted and updated by the Manifest GitHub Action.*

damien0x0023 committed:
telink: Add Telink tl7218 board (Tercel A1)

Added Telink tl7218 board.
Board is using RISC-V core refers SOC family telink_tlx:
- telink_tl721x

Signed-off-by: Damien Ji <yinghao.ji@telink-semi.com>
